x/mongo/driver/topology/connection.go

```diff
@@ -512,24 +513,53 @@ func (c *connection) close() error {
 }
 
 func (c *connection) closed() bool {
-	return atomic.LoadInt64(&c.state) == connDisconnected
+	if atomic.LoadInt64(&c.state) == connDisconnected {
```

Comment:
That's a great idea! This recommendation is also making me reconsider adding the logic to `connection.closed`, which previously would return quickly and do no system calls. It may be better to add a new method on `connection` or `pool` that is used in `connectionPerished` to prevent unintended side effects.

Reply:
Moved the new logic to a new method `connection.isAlive`.